src/encoded/views/search.py

```diff
@@ -334,8 +334,10 @@ def search(context, request, search_type=None):
         query['sort'] = get_sort_order()
         query['query']['match_all'] = {}
         del query['query']['query_string']
+        del query['highlight']
     elif len(doc_types) != 1:
         del query['query']['query_string']['fields']
+        del query['highlight']
```

Comment:
We want to keep the highlight behavior consistent irrespective of the size of the results.
If user see "highlights" for "limit=25" and doesn't for bigger "limits" then there will be a issue.

Though I will work around to improve performances in future release I think this should be good for now.